File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack1 = new Stack<>();

        for(String s: tokens){
            if(s.equals("+")){
                stack1.add(stack1.pop() + stack1.pop());
            }
            else if(s.equals("-")){
                int x = stack1.pop();
                stack1.add(stack1.pop() - x);
            }
            else if(s.equals("*")){
                stack1.add(stack1.pop() * stack1.pop());
            }
            else if(s.equals("/")){
                int x = stack1.pop();
                stack1.add(stack1.pop() / x);
            }
            else{
                stack1.add(Integer.valueOf(s));
            }
        }
        return stack1.pop();
    }
}

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for i in tokens:
            if i != "+" and i != "-" and i != "*" and i != "/":
                stack.append(int(i))
            
            else:

                num2 = stack.pop()
                num1 = stack.pop()
                result = 0

                if i == "+":
                    result = num1 + num2
                    stack.append(result)
                elif i == "-":
                    result = num1 - num2
                    stack.append(result)
                elif i == "*":
                    result = num1 * num2
                    stack.append(result)
                elif i == "/":
                    result = int(num1 / num2)
                    stack.append(result)

        return stack.pop()